Makefile

```
@@ -41,10 +42,19 @@ cloudprober-fips: $(SOURCES)
 	go tool nm cloudprober-fips | grep crypto/internal/boring/sig.BoringCrypto.abi0 > /dev/null || (echo "FIPS build failed: BoringCrypto not used" && rm cloudprober-fips && exit 1)
 	strip cloudprober-fips
 
+define make-binary-target-xds
+$1: $(SOURCES)
+	GOOS=$(subst macos,darwin,$(word 3,$(subst -, ,$1))) ; \
+	GOARCH=$(subst armv7,arm,$(word 4,$(subst -, ,$1))) ; \
+	GOARM=$(subst armv7,7,$(filter armv7,$(word 4,$(subst -, ,$1)))) ; \
+	CGO_ENABLED=0 GOOS=$$$${GOOS} GOARCH=$$$${GOARCH} GOARM=$$$${GOARM} go build -tags grpc_xds -o $1 -ldflags $(LDFLAGS) $(BINARY_SOURCE)
+endef
+
 test:
 	go test -v -race -covermode=atomic ./...
 
 $(foreach bin,$(BINARIES),$(eval $(call make-binary-target,$(bin))))
+$(foreach bin,$(BINARIES_XDS),$(eval $(call make-binary-target-xds,$(bin))))
```

You won't need separate target for xds binaries -- command line would determine which targets will be defined.
```suggestion
$(foreach bin,$(BINARIES),$(eval $(call make-binary-target,$(bin))))
```